### PR TITLE
Report runServer() failures better

### PR DESCRIPTION
**Pull request by kumar303**
This addresses https://github.com/mozilla/addons-frontend/issues/1682 but only in that it should show us what the real error was.

### CODE REVIEW

**Comment by kumar303** [comment]
@muffinresearch does this look ok?

File: src/core/server/base.js [hunk 1]
```diff
@@ -312,6 +311,7 @@ export function runServer({ listen = true, app = appName } = {}) {
     })
     .catch((err) => {
       log.error({ err });
+      throw err;
```
Comment (muffinresearch) [inline src/core/server/base.js]: What the rationale on re-throwing a caught error?

File: src/core/server/base.js [hunk 2]
```diff
@@ -269,9 +269,8 @@ function baseServer(routes, createStore, { appInstanceName = appName } = {}) {
 
 export function runServer({ listen = true, app = appName } = {}) {
   if (!app) {
-    log.fatal(
+    throw new Error(
       `Please specify a valid appName from ${config.get('validAppNames')}`);
-    process.exit(1);
```
Comment (muffinresearch) [inline src/core/server/base.js]: Is there a reason to throw and not bail at this point?

Comment (kumar303) [inline src/core/server/base.js]: I didn't want the test suite to be completely killed on this line: https://github.com/mozilla/addons-frontend/blob/master/tests/server/amo/TestDetailsPage.js#L16 I'd like it to fail with the error instead. Actually, I can't really tell if `process.exit(1)` was the right thing to do. When I add that back, `npm run dev:amo` just keeps spawning the process in a loop as if it's trying to start it again. Hmm.

File: src/core/server/base.js (hunk 1 above)
Comment (kumar303) [inline src/core/server/base.js]: Will it get logged elsewhere? I wanted to be sure it would.

File: src/core/server/base.js (hunk 2 above)
Comment (muffinresearch) [inline src/core/server/base.js]: I suppose that might be something that needs to change contextually. Having the process exit when the server is run in production due to a config error is likely to still be desirable.

File: src/core/server/base.js (hunk 1 above)
Comment (muffinresearch) [inline src/core/server/base.js]: I would have thought the logging should be enough. Another potentially relevant issue is the one for adding the sentry client which should help for tracking down caught and uncaught server exceptions. https://github.com/mozilla/addons-frontend/issues/1559

**Comment by kumar303** [comment]
@muffinresearch can you take another look? This puts back the `process.exit(1)` but for the server scripts only.

**Comment by muffinresearch** [comment]
Looks good to me.

**Pull request merged**